validphys2/src/validphys/filters.py

```diff
@@ -346,10 +425,29 @@ def _filter_closure_data(filter_path, data, fakepdf, fakenoise, filterseed, data
         output_cd = raw_cd.with_central_value(new_cv)
 
         # And export it to file
-        with open(data_path, "w", encoding="utf-8") as f:
-            output_cd.export_data(f)
-        with open(unc_path, "w", encoding="utf-8") as f:
-            output_cd.export_uncertainties(f)
+        output_cd.export_data(data_path.open("w", encoding="utf-8"))
```

Comment:
This has hit me recently in another context (heh), it is better to use the `with` here so you know not only that the file is closing (which will be when it gets out of scope anyway) but also that it doesn't close beforehand. Same below.